components/camel-coap/src/generated/resources/org/apache/camel/coap/coap+tcp.json

```diff
@@ -44,16 +44,16 @@
-    "privateKey": { "index": 12, "kind": "parameter", "displayName": "Private Key", "group": "security", "label": "security", "required": false, "type": "object", "javaType": "java.security.PrivateKey", "deprecated": false, "autowired": false, "secret": true, "description": "Set the configured private key for use with Raw Public Key." },
-    "pskStore": { "index": 13, "kind": "parameter", "displayName": "Psk Store", "group": "security", "label": "security", "required": false, "type": "object", "javaType": "org.eclipse.californium.scandium.dtls.pskstore.PskStore", "deprecated": false, "autowired": false, "secret": false, "description": "Set the PskStore to use for pre-shared key." },
-    "publicKey": { "index": 14, "kind": "parameter", "displayName": "Public Key", "group": "security", "label": "security", "required": false, "type": "object", "javaType": "java.security.PublicKey", "deprecated": false, "autowired": false, "secret": false, "description": "Set the configured public key for use with Raw Public Key." },
-    "recommendedCipherSuitesOnly": { "index": 15, "kind": "parameter", "displayName": "Recommended Cipher Suites Only", "group": "security", "label": "security", "required": false, "type": "boolean", "javaType": "boolean", "deprecated": false, "autowired": false, "secret": false, "defaultValue": true, "description": "The CBC cipher suites are not recommended. If you want to use them, you first need to set the recommendedCipherSuitesOnly option to false." },
-    "sslContextParameters": { "index": 16, "kind": "parameter", "displayName": "Ssl Context Parameters", "group": "security", "label": "security", "required": false, "type": "object", "javaType": "org.apache.camel.support.jsse.SSLContextParameters", "deprecated": false, "autowired": false, "secret": false, "description": "Set the SSLContextParameters object for setting up TLS. This is required for coapstcp, and for coaps when we are using certificates for TLS (as opposed to RPK or PKS)." },
-    "trustedRpkStore": { "index": 17, "kind": "parameter", "displayName": "Trusted Rpk Store", "group": "security", "label": "security", "required": false, "type": "object", "javaType": "org.eclipse.californium.scandium.dtls.rpkstore.TrustedRpkStore", "deprecated": false, "autowired": false, "secret": false, "description": "Set the TrustedRpkStore to use to determine trust in raw public keys." }
+    "advancedCertificateVerifier": { "index": 9, "kind": "parameter", "displayName": "Advanced Certificate Verifier", "group": "security", "label": "security", "required": false, "type": "object", "javaType": "org.eclipse.californium.scandium.dtls.x509.NewAdvancedCertificateVerifier", "deprecated": false, "autowired": false, "secret": false, "description": "Set the AdvancedCertificateVerifier to use to determine trust in raw public keys." },
```

Comment:
TCP/TLS support is still "experimental". It doesn't share too much with the DTLS possibilities. The configuration properties are defined in [TcpConfig](https://github.com/eclipse-californium/californium/blob/main/element-connector/src/main/java/org/eclipse/californium/elements/config/TcpConfig.java).

The `AdvancedCertificateVerifier` is only supported for DTLS. 
For TLS the standard java SSLContext is used. As long as you don't want to use remote-calls to fetch/verify certificates, that works pretty well and is very common for java.

Reply:
Thanks @boaks . The JSON files and `CoapEndpointConfigurer` are automatically generated.

From what I can see `AdvancedCertificateVerifier` and `PSK` are only used in the DTLS configuration setup ([here](https://github.com/apache/camel/pull/12769/files#diff-fa9b8056c9aaac49ce4368ccadc8e7d9748f3dbdb1d0a59481b8cecdbf8d6cc7R424)).